### ISSUE TITLE
MaxSuggestionInViewport does not display correctly

### ISSUE THREAD

**Issue by bebaoboy** [maheshj01/searchfield#152]
**Describe the bug**
There are some bugs when setting the attribute `maxSuggestionsInViewport`.

**To Reproduce**
## 1. Incorrect Display
I have a simple search field like this with about 16 items. `maxSuggestionsInViewPort` is set to 5 by default.
![image](https://github.com/user-attachments/assets/7bc104be-2694-4eae-88f8-69a2b4bafc76)
However, the suggestion result in the app does not show 5 item.
![image](https://github.com/user-attachments/assets/57c4edb6-360f-4bd8-ab5b-5d16cdabf78a)

<br>

## 2. Does not have a lower bound like it says in the document.
![image](https://github.com/user-attachments/assets/a752ef71-7872-4e11-9d56-170abc396a6c)
But I can still set it to a value smaller than 5. 
![image](https://github.com/user-attachments/assets/5b0d8f7c-3a4f-4d4a-b2a0-e7ae32b039de)
![image](https://github.com/user-attachments/assets/b54f58ed-96d8-49a2-b72a-86bfd61ca668)
Again the `maxSuggestionsInViewPort` display incorrectly.
- If I set it to 0, the whole suggestion box disappear completely.
- If I set to a negative value, well, it **crashes**!

### Suggested Solution: add a lowerbound check.

<br>

## 3. Upper bound?
So I try to set the value to 16, which is the number of my items (supposed to show JKL10 ... as well). This also happens with value that is bigger than the number of items.
Result: the suggestion box does not show all my item. Is this an expected behaviour?
![image](https://github.com/user-attachments/assets/972366fc-451b-466b-bb88-43d590ab9f6e)

<br>
[x] By clicking this checkbox, I confirm I am using the latest version of the package found on [pub.dev/searchfield](https://pub.dev/packages/searchfield) and from master branch on this repo. No library file has been edited.

**Code sample**
<details>
<summary>Show code sample</summary>

```dart  
SearchField(
  hint: 'Basic SearchField',
  maxSuggestionsInViewPort: 16,    //<--- trouble
  initialValue: SearchFieldListItem<String>('ABC'),
  suggestions: [
    'ABC',
    'DEF',
    'GHI',
    'JKL',
    'JKL1',
    'JKL2',
    'JKL3',
    'JKL4',
    'JKL5',
    'JKL6',
    'JKL7',
    'JKL8',
    'JKL9',
    'JKL10',
    'JKL11',
    'JKL12',
  ].map(SearchFieldListItem<String>.new).toList(),
  suggestionState: Suggestion.expand,
),
```
</details>

- @maheshmnj, I was doing the #67 when I stumbled upon this so I have to stop and report this issue. Please have a look. Thank you!

**Comment by maheshmnj** [comment]
Thanks for catching that, we definetly need to fix that logic and have basic sanity checks such as zero or negative height.

**Comment by bebaoboy** [comment]
@maheshmnj can you please point me in a direction for the first bug (most important one) listed above? I'm not sure where or how that happened in the code. Thank you.

**Comment by maheshmnj** [comment]
@bebaoboy 

Just adjust the `itemHeight` and `maxSuggestionsInViewPort`

```dart
Form(
  key: _formKey,
  child: SearchField(
    suggestions: _statesOfIndia
        .map((e) => SearchFieldListItem(e))
        .toList(),
    suggestionState: Suggestion.expand,
    textInputAction: TextInputAction.next,
    hint: 'SearchField Example 2',
    searchStyle: TextStyle(
      fontSize: 18,
      color: Colors.black.withOpacity(0.8),
    ),
    validator: (x) {
      if (!_statesOfIndia.contains(x) || x!.isEmpty) {
        return 'Please Enter a valid State';
      }
      return null;
    },
    searchInputDecoration: InputDecoration(
      focusedBorder: OutlineInputBorder(
        borderSide: BorderSide(
          color: Colors.black.withOpacity(0.8),
        ),
      ),
      border: OutlineInputBorder(
        borderSide: BorderSide(color: Colors.red),
      ),
    ),
    maxSuggestionsInViewPort: 6, // adjust the number of Suggestions
    itemHeight: 50,  // adjusts each suggestionsHeight
    onTap: (x) {},
  ))
```

**Comment by bebaoboy** [comment]
@maheshmnj sorry, i mean the code where the suggestion box height is calculating. Because in the first bug, I set maxHeight in view port = 5 but the box shows only 4 suggestions...

**Comment by maheshmnj** [comment]
This is where it is buggy https://github.com/maheshmnj/searchfield/blob/2925d80f9136c3bf1e873367c5a0ad33163575f8/lib/src/searchfield.dart#L628

**Comment by maheshj01** [comment]
Fixed and Released in [v1.0.7](https://pub.dev/packages/searchfield/changelog)
